src/utils/components/DiskModal/HTTPDiskModal.tsx

```diff
@@ -0,0 +1,103 @@
+import React, { FC } from 'react';
+import { FormProvider, useForm } from 'react-hook-form';
+
+import { useKubevirtTranslation } from '@kubevirt-utils/hooks/useKubevirtTranslation';
+import { OS_NAME_TYPES } from '@kubevirt-utils/resources/template';
+import { getBootDisk } from '@kubevirt-utils/resources/vm';
+import { hasSizeUnit as getOSNameWithoutVersionNumber } from '@kubevirt-utils/resources/vm/utils/disk/size';
+import { isEmpty } from '@kubevirt-utils/utils/utils';
+import { Form } from '@patternfly/react-core';
+import { isRunning } from '@virtualmachines/utils';
+
+import { PendingChangesAlert } from '../PendingChanges/PendingChangesAlert/PendingChangesAlert';
+import TabModal from '../TabModal/TabModal';
+
+import AdvancedSettings from './components/AdvancedSettings/AdvancedSettings';
+import BootSourceCheckbox from './components/BootSourceCheckbox/BootSourceCheckbox';
+import DiskInterfaceSelect from './components/DiskInterfaceSelect/DiskInterfaceSelect';
+import DiskNameInput from './components/DiskNameInput/DiskNameInput';
+import DiskSizeInput from './components/DiskSizeInput/DiskSizeInput';
+import DiskSourceUrlInput from './components/DiskSourceSelect/components/DiskSourceUrlInput/DiskSourceUrlInput';
+import DiskTypeSelect from './components/DiskTypeSelect/DiskTypeSelect';
+import StorageClassAndPreallocation from './components/StorageClassAndPreallocation/StorageClassAndPreallocation';
+import { getDefaultCreateValues, getDefaultEditValues } from './utils/form';
+import { checkDifferentStorageClassFromBootPVC, diskModalTitle, getOS } from './utils/helpers';
+import { submit } from './utils/submit';
+import { SourceTypes, V1DiskFormState, V1DiskModalProps } from './utils/types';
+
+const HTTPDiskModal: FC<V1DiskModalProps> = ({
+  editDiskName,
+  isCreated,
+  isOpen,
+  onClose,
+  onSubmit,
+  vm,
+}) => {
+  const { t } = useKubevirtTranslation();
+
+  const os = getOS(vm);
+  const isVMRunning = isRunning(vm);
+
+  const isEditDisk = !isEmpty(editDiskName);
+
+  const initialBootSource = getBootDisk(vm)?.name;
+
+  const isInitialBootDisk = initialBootSource === editDiskName;
+
+  const methods = useForm<V1DiskFormState>({
+    defaultValues: isEmpty(editDiskName)
+      ? getDefaultCreateValues(vm, SourceTypes.HTTP)
+      : getDefaultEditValues(vm, editDiskName),
+    mode: 'all',
+  });
```

Comment:
this repeats a lot, this be extracted to a hook?

Reply:
i changed a little bit the boot methods. Calling them directly in the BootSourceCheckbox component